scripts/zkai_d128_proof_native_two_slice_compression_gate.py

```
@@ -0,0 +1,1088 @@
+            raise D128ProofNativeTwoSliceCompressionError(f"mutation case {index} error must be a string")
+        if rejected_flag:
+            rejected += 1
+    expect_equal(tuple(pairs), EXPECTED_MUTATION_INVENTORY, "mutation case inventory")
+    expect_equal(payload.get("case_count"), len(cases), "case_count")
+    expect_equal(payload.get("all_mutations_rejected"), all(case["rejected"] for case in cases), "all_mutations_rejected")
+    expected_by_pair = {(case["mutation"], case["surface"]): case for case in mutation_cases(_draft_payload_for_case_replay(payload))}
+    for index, raw_case in enumerate(cases):
+        case = require_object(raw_case, f"mutation case {index}")
+        expected = expected_by_pair.get((case["mutation"], case["surface"]))
+        if expected is None:
+            raise D128ProofNativeTwoSliceCompressionError(f"missing recomputed mutation case {index}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)